packages/patterns/src/components/category-selector.js

```diff
@@ -92,13 +95,21 @@ export default function CategorySelector( { onCategorySelection } ) {
 
 		setValues( uniqueTerms );
 
-		Promise.all(
-			uniqueTerms.map( ( termName ) =>
-				findOrCreateTerm( { name: termName } )
-			)
-		).then( ( newTerms ) => {
-			onCategorySelection( newTerms );
+		// If the user clicks the create pattern modal button directly after entering
+		// a category we need to return a promise so the pattern doesn't save before
+		// the save of the categories is completed.
+		const categorySaving = new Promise( function ( resolve ) {
+			Promise.all(
+				uniqueTerms.map( ( termName ) =>
+					findOrCreateTerm( { name: termName } )
+				)
+			).then( ( newTerms ) => {
+				setCategorySaving();
+				onCategorySelection( newTerms );
+				resolve( newTerms );
+			} );
```

Comment:
I think we can just chain the promise, making it a bit easier to read.

```suggestion
		const categorySaving = Promise.all(
			uniqueTerms.map( ( termName ) =>
				findOrCreateTerm( { name: termName } )
			)
		).then( ( newTerms ) => {
			setCategorySaving();
			onCategorySelection( newTerms );
			return newTerms ;
		} );
```

Reply:
hmm, yeh, that extra promise wrapper was a bit redundant, thanks - have removed.